### PR TITLE
Fix session threshold check to be compatible w/ tinygo

### PR DESCRIPTION
**Pull request by IcanDivideBy0**

### CODE REVIEW

File: internal/round/helper.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func NewSession(info Info, sessionID []byte, pl *pool.Pool, auxInfo ...hash.Writ
 	}
 
 	// make sure the threshold is correct
-	if info.Threshold < 0 || info.Threshold > math.MaxUint32 {
+	if info.Threshold < 0 || info.Threshold > math.MaxInt32 {
```
Comment (Konubinix) [inline internal/round/helper.go:55]: Can you ellaborate on why this is related to tinygo

The following program runs on tinygo seamlessly

```
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(math.MaxUint32)
}
```

So I suppose this is a mistake. Isn't it?

Comment (IcanDivideBy0) [reply internal/round/helper.go:55]: try the following :
```go
package main

import (
	"fmt"
	"math"
)

func main() {
	var a int = 12
	fmt.Println(a > math.MaxUint32)
}
```

```sh
tinygo build -target wasm main.go
```

Comment (Konubinix) [reply internal/round/helper.go:55]: Of course, you are comparing a signed int with an unsigned int,

fix this with 

```
package main

import (
	"fmt"
	"math"
)

func main() {
	var a uint32 = 12
	fmt.Println(a > math.MaxUint32)
}
```

That does not tell why using a signed int makes sense related to tinygo in your example. If seems more sensible to me to set the Threshold type in the Info struct to a uint.

**Comment by Konubinix** [comment]
After discussing with @IcanDivideBy0, I understood that there is no work planned on this yet. I'm closing until that situation changes.

**Pull request closed without merging**